### PR TITLE
[NTOS:EX] Fix handle table code for x64.

### PR DESCRIPTION
**Pull request by tkreuzer**
Based on patch by Ivan Labutin. See PR #115

### CODE REVIEW

File: ntoskrnl/include/internal/ex.h [hunk 1]
```diff
@@ -68,25 +68,29 @@ VOID NTAPI ExpDebuggerWorker(IN PVOID Context);
...
 
 typedef union _EXHANDLE
 {
      struct
      {
-         ULONG_PTR TagBits:2;
-         ULONG_PTR Index:29;
+         ULONG_PTR TagBits:     HANDLE_TAG_BITS;
+         ULONG_PTR Index:       HANDLE_INDEX_BITS;
+         ULONG_PTR KernelFlag : KERNEL_FLAG_BITS;
```
Comment (ThFabba) [inline ntoskrnl/include/internal/ex.h]: `C_ASSERT(KERNEL_FLAG_BITS + HANDLE_INDEX_BITS + HANDLE_TAG_BITS == 8*sizeof(PVOID));` might be useful

Comment (ThFabba) [reply ntoskrnl/include/internal/ex.h]: Well, I guess the new KERNEL_FLAG_BITS definition already takes care of that...

File: ntoskrnl/ex/handle.c [hunk 2]
```diff
@@ -18,6 +18,7 @@
 LIST_ENTRY HandleTableListHead;
 EX_PUSH_LOCK HandleTableListLock;
 #define SizeOfHandle(x) (sizeof(HANDLE) * (x))
+#define INDEX_TO_HANDLE_VALUE(x) ((x) << 2)
```
Comment (ThFabba) [inline ntoskrnl/ex/handle.c]: Should that 2 be HANDLE_TAG_BITS?

File: ntoskrnl/ex/handle.c [hunk 3]
```diff
@@ -18,6 +18,7 @@
 LIST_ENTRY HandleTableListHead;
 EX_PUSH_LOCK HandleTableListLock;
 #define SizeOfHandle(x) (sizeof(HANDLE) * (x))
```
Comment (HBelusca) [inline ntoskrnl/ex/handle.c]: Is "SizeOfHandle" still used in the code? (since I see many replacements of SizeOfHandle to INDEX_TO_HANDLE_VALUE)

**Pull request merged**